src/wasm.h

```diff
@@ -1615,8 +1604,11 @@ class ArrayNewSeg : public SpecificExpression<Expression::ArrayNewSegId> {
 public:
   ArrayNewSeg(MixedArena& allocator) {}
 
-  ArrayNewSegOp op;
-  Name segment;
+  // One of the two should be defined, which specifies if this is an
+  // array.new_data or array.new_elem.
```

Comment:
I do think it would make more sense to split this into separate `ArrayNewData` and `ArrayNewSeg` rather than having two different names. Almost all the visitors do different (but similar) things for the two operations, so I think this would increase complexity less than having these mutually exclusive fields.